### PR TITLE
posialert console, give posibrains a chance

### PR DESCRIPTION
**Pull request by jjpark-kb**
[tested]
added the posialert console to many maps
[roid][packed][meta][box][taxi]
:cl:
 * rscadd: added the posialert console. A console that allows ghosts to (on a cooldown) notify nearby players they wish to play as a posibrain. If ghosts are unruly, living crewmembers can turn off the posialert console.

### CODE REVIEW

**Comment by jjpark-kb** [comment]
I honestly don't know all the maps that I need to modify (or not modify). So odds are, I probably missed some tags as well, sorry!

**Comment by StreptoMutans** [comment]
The posibrains ping when spooked, the borer egg wiggles and for dionae there's literally nothing but messing with botany lights.
Why favor posibrains out of all options to return into the game?

**Comment by jjpark-kb** [comment]
> The posibrains ping when spooked, the borer egg wiggles and for dionae there's literally nothing but messing with botany lights.
> Why favor posibrains out of all options to return into the game?

So let's quickly talk about the reasons for this posi-alert console. The main use of it will be after all posibrains are gone and another ghost wishes to become a posibrain. So they (the ghost) will use the console and it will notify people in sight range that there are "personalities" willing to be downloaded. This basically boils down to "Hey, create more posibrains." So now lets talk about me "favoring" the posibrains.

Borers start with eggs already, similar to the posibrains; however, the means of creating eggs is far different. Compared to the printing of the posibrains, borers require already active (and alive) borers to create more eggs, no? That means that ghosts just have to rely on borers being smart and creating new children for them to play as borers.

Going to be honest, never played dionae and have no idea about the gameplay about them. If there is a solid case for why there should be a console for them, It would be really easy.

**Comment by SonixApache** [comment]
i really don't see the point for this

**Comment by PlausibleDeniability** [comment]
Would it not be simpler to just alert science department PDAs when a posibrain is pinged?

**Comment by jjpark-kb** [comment]
> Would it not be simpler to just alert science department PDAs when a posibrain is pinged?

again, this is for once all posibrains have been used. This is for when ghosts want the crew to make more posibrains.

**Comment by PlausibleDeniability** [comment]
Ah my bad, gotcha

**Comment by PrimeDSS13** [comment]
I'm not sure if it needs its own console,  I kind of like Plausibles idea of it being a PDA thing. Maybe have it be a feature of the science PDA, where it gives an alert when a ghost requests it. (with a button to turn it off)

**Comment by jjpark-kb** [comment]
> I'm not sure if it needs its own console, I kind of like Plausibles idea of it being a PDA thing. Maybe have it be a feature of the science PDA, where it gives an alert when a ghost requests it. (with a button to turn it off)

Its a wall-mount, same naming as the request console.
image

**Comment by PrimeDSS13** [comment]
oh yeah that's fine then

**Comment by beesil** [comment]
`posialert1` state has a stray pixel in the top left corner

**Comment by jjpark-kb** [comment]
> `posialert1` state has a stray pixel in the top left corner

good catch, thank you, will fix.
image

okay, I think the icon diff bot is just messing up or something

**Comment by jjpark-kb** [comment]
it's a blessing that this is still not conflicting with the other PRs that had mapping changes.

**Comment by Kurfursten** [comment]
It does not look like xoq was saved in tgm format.

File: code/game/machinery/posialert_console.dm [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	if(ghost_cooldown > world.time)
+		to_chat(user, "\The [src] is still on cooldown, come back later!")
+		return ..()
+	ghost_cooldown = world.time + 1 MINUTES //hopefully 1 minute is enough. at least its adjustable!
+	log_game("[key_name(user)] has activated the posi-alert console.")
+	visible_message("<span class='notice'>There are positronic personalities available for download!</span>")
+	playsound(get_turf(src), 'sound/machines/ping.ogg', 50, 1)
+	flick("posialert1",src)
+
+/obj/machinery/posialert_console/proc/update_ghost_overlay()
```
Comment (Kurfursten) [inline code/game/machinery/posialert_console.dm:57]: why not as part of the usual update_icon()?

File: code/game/machinery/posialert_console.dm [hunk 2]
```diff
@@ -0,0 +1,60 @@
+/*************************
+*** Posi-Alert Console ***
+*************************/
+
+/obj/machinery/posialert_console
```
Comment (Kurfursten) [inline code/game/machinery/posialert_console.dm:5]: any reason to not make this a computer? no way to construct or deconstruct either...

Comment (jjpark-kb) [reply code/game/machinery/posialert_console.dm:5]: as others have said, its really a small function to be made into a computer.
Ill add a way to construct, deconstruct may be a little harder.

File: code/game/machinery/posialert_console.dm (hunk 1 above)
Comment (jjpark-kb) [reply code/game/machinery/posialert_console.dm:57]: will do.

**Comment by jjpark-kb** [comment]
I have become busy with college, so I really can't use my time to fix or continue this PR. Sorry :(

**Pull request closed without merging**